pixmappy/PixelMapCollection.py

```diff
@@ -374,33 +411,49 @@ def __init__(self, name, pmap, projection, scale=1.):
         self.projection = projection
         self.scale = scale
         self.dest = None
-        return
+
     def reprojectTo(self, projection):
         self.dest = projection
-        return
-    def toSky(self, xy, c=None):
+
+    def toSky(self, x, y, c=None):
         ''' Return sky coordinates corresponding to array
         '''
-        return self.projection.toSky(self.pmap(xy,c) * self.scale)
-    def toPix(self, coords, c=None, guess=np.array([0.,0.])):
-        ''' Return pixel coordinates corresponding to input SkyCoords.
+        # This is the front end user interface.  Let's not modify the user's input x,y values.
+        # Everything else is allowed to modify the inputs to make the outputs more efficiently.
+        x1 = np.array(x, dtype=float)
+        y1 = np.array(y, dtype=float)
+        xw, yw = self.pmap(x1,y1,c)
+        xw *= self.scale
+        yw *= self.scale
+        return self.projection.toSky(xw, yw)
+
+    def toPix(self, ra, dec, c=None, guess=np.array([0.,0.])):
+        ''' Return pixel coordinates corresponding to input RA, Dec.
         guess is a starting guess for solver, which can be either a single
         coordinate pair or an array matching coords length.
         '''
-        xyw = self.projection.toXY(coords) / self.scale
-        xyp = np.zeros_like(xyw) + guess  # Broadcasts to same dimensions as input
-        self.pmap.inverse(xyw, xyp, c)  # ??? tolerance???
-        return xyp
-    def __call__(self,xy,c=None):
+        xw, yw = self.projection.toXY(ra, dec)
+        xw /= self.scale
+        yw /= self.scale
+        xp = np.empty_like(xw)
+        yp = np.empty_like(yw)
+        xp.fill(guess[0])
+        yp.fill(guess[1])
+        xp, yp = self.pmap.inverse(xw, yw, xp, yp, c)  # ??? tolerance???
+        return xp, yp
+
+    def __call__(self, x, y, c=None):
         '''Map the input coordinates to the coordinates either in the original
         projection, or in the reprojected system if one has been given.
         '''
-        xyw = self.pmap(xy,c)
+        x,y = self.pmap(x,y,c)
         if self.dest is not None:
             # Execute reprojection
-            coords = self.projection.toSky(xyw*self.scale)
-            xyw = self.dest.toXY(coords) / self.scale
-        return xyw
+            coords = self.projection.toSky(x*self.scale, y*self.scale)
+            x, y = self.dest.toXY(coords)
```

Comment:
Yes.  Which tells me that none of the current unit tests hit this branch.  Probably ought to add something that does a reprojection.  Anyway, [fixed](7a1f322)